packages/protvista-structure/src/protvista-structure.js

```diff
@@ -103,58 +112,87 @@ class ProtvistaStructure extends HTMLElement {
     return this.setAttribute("accession", accession);
   }
 
-  get isResidueOnlyHighlight() {
-    return this.hasAttribute("highlightresidues");
+  get hideTable() {
+    return this.getAttribute("hide-table");
+  }
+
+  get molecule() {
+    return this.getAttribute("molecule");
+  }
+
+  get height() {
+    return this.getAttribute("height");
+  }
+
+  get hideViewportControls() {
+    return this.getAttribute("hide-viewport-controls");
   }
 
   connectedCallback() {
     if (this.closest("protvista-manager")) {
       this.manager = this.closest("protvista-manager");
       this.manager.register(this);
     }
+
     const style = document.createElement("style");
-    style.innerHTML = ProtvistaStructure.css;
+    style.innerHTML = this.css;
     this.appendChild(style);
-    this.titleContainer = document.createElement("h4");
+
     const flexContainer = document.createElement("div");
     flexContainer.className = "main-container";
-    this.titleContainer.id = "litemol-title";
-    this.tableDiv = document.createElement("div");
-    this.tableDiv.className = "table-container";
     const litemolDiv = document.createElement("div");
     litemolDiv.className = "litemol-container";
     litemolDiv.id = "litemol-instance";
     this.messageContainer = document.createElement("span");
-    this.appendChild(this.titleContainer);
     this.appendChild(this.messageContainer);
     this.appendChild(flexContainer);
     flexContainer.appendChild(litemolDiv);
-    flexContainer.appendChild(this.tableDiv);
+
+    if (!this.hideTable) {
+      this.tableDiv = document.createElement("div");
+      this.tableDiv.className = "table-container";
+      flexContainer.appendChild(this.tableDiv);
+    }
+
     this.loadLiteMol();
-    this.loadUniProtEntry().then(entry => {
-      this._pdbEntries = entry.dbReferences
-        .filter(dbref => dbref.type === "PDB")
-        .map(d => {
-          return {
-            id: d.id,
-            properties: {
-              method: ProtvistaStructure.formatMethod(d.properties.method),
-              chains: ProtvistaStructure.getChains(d.properties.chains),
-              resolution: ProtvistaStructure.formatAngstrom(
-                d.properties.resolution
-              )
-            }
-          };
-        });
-      if (this._pdbEntries.length <= 0) {
-        this.style.display = "none";
-        return;
-      }
-      this.loadStructureTable();
-      this.selectMolecule(
-        this._pdbEntries.filter(d => d.properties.method !== "Model")[0].id
-      );
-    });
+
+    if (!this.hideTable || !this.molecule) {
```

Comment:
There is a case where we both the table to be loaded from UniProt AND `molecule` to be specified. I don't think this case is being taken care of?

Reply:
I'm not sure if I follow. This `if` says if we are not hiding the table or we have passed a molecule, then load the required data from UniProt. Wouldn't that cover the use-case since the molecule is provided?